src/averageAges.js

```diff
@@ -54,8 +72,20 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const women = people.filter(person =>
+    person.sex === 'f'
+    && people.some(p => p.mother === person.name
+    && (onlyWithSon ? p.sex === 'm' : true)));
+
+  const ages = women.map(w => people.filter(child => child.mother === w.name
+    && (onlyWithSon ? child.sex === 'm' : true)).map(ch =>
+    ch.born - w.born)).flatMap(t => t);
+
+  const sum = ages.reduce((a, b) => a + b);
```

Comment:
Looks too complicated,
You can try smth like
` const children = people.filter(child => people.find(mother =>
    child.mother === mother.name) && (!onlyWithSon || child.sex === 'm')`
    
    Then find ageDifference using reduce method and then calculate average age
    
    If you have any questions - feel free to ask them in fe_chat